.jenkins/pipeline.jenkins

```diff
@@ -0,0 +1,686 @@
+//
+//
+// Syntax help can be found at https://ci-cassandra.apache.org/pipeline-syntax/
+//
+// Validate/lint this file using the following command
+// `curl -X POST  -F "jenkinsfile=<.jenkins/Jenkinsfile" https://ci-cassandra.apache.org/pipeline-model-converter/validate`
+
+pipeline {
+  agent {
+    kubernetes {
+      label 'cassandra'
+      defaultContainer 'jnlp'
+      yaml '''
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          labels:
+            jenkins: label
+        spec:
+          initContainers:
+            - name: init-docker-socket
+              image: alpine:latest
+              imagePullPolicy: Always
+              command:
+                - sh
+                - -c
+                - chown 1000:1000 /var/run/docker.sock
+              volumeMounts:
+                - name: docker-socket
+                  mountPath: /var/run/docker.sock
+              securityContext:
+                runAsUser: 0
+          containers:
+          - name: jnlp
+            image: richamishra006/apache-k8s-agent:32
+            command:
+              - /bin/sh 
+              - -c
+              - |
+                /usr/local/bin/jenkins-agent
+            volumeMounts:
+            - name: docker-socket
+              mountPath: /var/run/docker.sock
+            - name: cassandra-dir
+              mountPath: /tmp/k8s-e2e
+            tty: true
+            securityContext:
+              runAsUser: 0
+              privileged: true
+          nodeSelector:
+            gke-node: first
+          volumes:
+          - name: docker-socket
+            hostPath:
+              path: /var/run/docker.sock
+          - name: cassandra-dir
+            hostPath:
+              path: /home/jenkins/agent/workspace/k8s-e2e
+        '''
+    }
+  }
+
+  options {
+    githubProjectProperty('https://github.com/apache/cassandra')
+    parallelsAlwaysFailFast()
+    timestamps()
+    //skipDefaultCheckout()
+  }
+  environment {
+    javaVersionDefault = javaVersionDefault()
+    javaVersionsSupported = javaVersionsSupported()
+    PIPELINE_SCRIPT_DEBUG = 'true'
+  }
+  stages {
+    stage('Init') {
+      steps {
+        container('jnlp') {
+          //cleanWs()
+          script { currentBuild.result='SUCCESS' }
+          sh 'echo "stage init"'
+          }
+      }
+    }
+
+    stage('Builds') {
+      parallel {
+        stage('artifacts') {
+          steps {          
+            sh 'echo "starting artifacts stage"'
+            //sh 'sleep 3000'
+            //sh 'cp -r /tmp/* /home/jenkins/agent/workspace/k8s-e2e'
+            //script { env.javaVersionDefault = javaVersionDefault() }
+            //script { env.javaVersionsSupported = javaVersionsSupported() }
+            buildJob(".build/docker/_docker_run.sh bullseye-build.docker build-jars.sh", "apache-cassandra-*.tar.gz,apache-cassandra-*.jar,apache-cassandra-*.pom")
```

Comment:
this will need to be changed back 
```
            buildJob(".build/docker/build-artifacts.sh", "apache-cassandra-*.tar.gz,apache-cassandra-*.jar,apache-cassandra-*.pom")
```
and this will test whether our stash is only grabbing what it needs